### DISCUSSION TITLE
Garage door

### DISCUSSION THREAD

**Discussion by AndWik** [Thomas55555/husqvarna_automower#247] [Ideas]
My long time wish is to have a garage door. It should open ONLY when the mower is starting or the mower is returning. This integration allows for the latter.
Using "Leaving" does not help as when it ever happens it happens far too late.
Beeing able to use "Next start" and calculate time it may open up to 15 minutes before.

Any idea how to implement this would be very appreciated

Mr Anders Wikstrom
Sweden

**Comment by prairiesnpr** [comment]
Maybe something like this?
`{{ (utcnow() >= ( states('sensor.aeinarr_next_start')|as_datetime - timedelta(minutes=15))) and (utcnow() < ( states('sensor.aeinarr_next_start')|as_datetime + timedelta(minutes=15))) }}`

**Comment by mulcmu** [comment]
`sensor.automower_next_start` changes to `unknown` when it starts.  On my 430XH `sensor.automower_next_start` updates a few times as the battery approaches full charge.  I haven't paid too much attention to how accurate this estimate is.  The transition to `unknown` takes place at the same time as "leaving" and transition to "cleaning" so likely "far too late" as well.

Could the garage door opening controller be setup with a switch or proximity sensor to detect the mower starting to leave the dock and open the door in time?

Another option could be using a schedule override to park mower while charging.  Once charged automation could open the garage door and remove the park override.

**Reply by prairiesnpr** [reply]
I think the best way to approach this, other than the manual option that most  people go with, is to put the mower in eco mode and then monitor the loop circuit.  Since the mower will power the loop circuit immediately before it leaves the dock, that probably the best signal to open a door on.

**Reply by AndWik** [reply]
This approach attracts me a lot. Do you sense the connector electrically or use a coil?
When going home. Is the a way to read from status? Non of the selections in UI allows me to select "RETURNING_HOME"

**Reply by prairiesnpr** [reply]
The source I read said that it could be done by measuring the voltage on the line, said it was around 30V.  I have not tested on my 450XH.

As far as a return notification, I'm using the vacuum entity as shown here. https://github.com/prairiesnpr/Home-AssistantConfig/blob/master/automations/mower/notify_on_mower_returning.yaml
